File: codes/jiheon/209_minimumsizesubarraysum.rs
impl Solution {
    pub fn min_sub_array_len(target: i32, nums: Vec<i32>) -> i32 {
        let mut left = 0usize;
        let mut sum = 0;
        let mut ans = usize::MAX;

        for (i, num) in nums.iter().enumerate() {
            sum += num;

            while sum - nums[left] >= target {
                sum -= nums[left];
                left += 1;
            }

            if sum >= target {
                ans = ans.min(i - left + 1);
            }
        }

        if ans == usize::MAX {
            0
        } else {
            ans as _
        }
    }
}

File: codes/naeun/209_minimumsizesubarraysum.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int result = Integer.MAX_VALUE;

        // do double iteration 
        // if sum matches target, update result with min value
        for(int left = 0; left < nums.length; left++){
            int temp = 0;
            for(int right = left; right < nums.length; right++){
                temp += nums[right];

                if(temp >= target){
                    result = Math.min(result, right-left+1);
                    break;
                }
            }
        }

        if(result == Integer.MAX_VALUE){
            return 0;
        }
        return result;
    }
}